src/browser/js/main.js

```diff
@@ -19,7 +19,145 @@ import $ from 'jquery';
+    }))
+}
+
+const addNewToDo = async (params) => {
+    const response = await fetch('https://jsonplaceholder.typicode.com/todos',
+                {
+                    method: 'POST',
+                    body: JSON.stringify({title: params.title})
+                });
+    if (response.ok)
+    {
+       createToDo({ title: params.title });
+    }
+    /*const result = await response.json();
+    console.log(result.message);
+    alert(result.message);*/
+}
+
+const deleteToDo = (element) => {
+    if (element.matches('[data-todo-id]')) {
+        if (element.closest('[data-list-todo-id]')) {
+            element.closest('[data-list-todo-id]').remove();
+        }
+    }
+}
+
+const filterToDolist = (item) => {
+    const allToDo = document.querySelectorAll('[data-list-todo-id]');
+    allToDo?.forEach((element) => {
+        for (const child of element.children)
+        {
+            if (child.matches('span'))
+            {
+                if (String(child.innerHTML).indexOf(item.value) >= 0)
+                {
+                  //console.log(element);
+                  element.classList.remove('filtred');
+                }
+                else
+                {          
+                    console.log(element);
+                    //element.remove();
+                   //element.style
+                   //console.log(child.innerHTML);
+                  element.classList.add('filtred');
+                };
+            }
+        }
+        
+     })
+    ///console.log(element.value);
+    /*
+    }*/
+}
+
+const loadMoreToDo = () => {
+    getToDoFetch()
+        .then((ToDo) => ToDo?.forEach(element => {
+            createToDo(element);
+        } ) )
+        .catch((error) => console.log(error))
+        .finally()
+
+}
+
+const initPageEvents = () => {
+    const refreshButton = document.getElementById('btn-refresh');
+    const inputBlock = document.getElementById('form-input-js');
+    const listToDo = document.getElementById('list-group-todos');
+    const inputFilter = document.getElementById('form-search-js');
+
+    if (inputFilter)
+    {
+        inputFilter.addEventListener('submit', (event) => {
+            event.preventDefault();
+            filterToDolist(event.currentTarget.elements.search);
+        });
+    }
+
+    if (listToDo) {
+        listToDo.addEventListener('click', (event) => deleteToDo(event.target));
+    };
+
+    if (inputBlock) {
+        inputBlock.addEventListener('submit', (event) => {
+            event.preventDefault();
+            const InputText = event.currentTarget.elements.add
+            addNewToDo({ title: InputText.value });
+            InputText.value = '';
+        });
+    };
+
+    if (refreshButton)
+    {
+        refreshButton.addEventListener('click', (event) => {
+            event.target.classList.add('refresh-button--loading');
+            loadMoreToDo(event.target);
+            event.target.classList.remove('refresh-button--loading');
```

Comment:
Вот с таким решением кнопка никогда не будет крутится, потому что окончание загрузки произойдет только в функции loadMoreTodo после разрешения или реджекта промиса, который возвращает getTodoFetch

Можешь поставить консоль логи на местах, где ты добавляешь и убираешь класс, и ещё один консоль лог внутри then, который вызывается от getTodoFetch. Так ты увидишь что первые два вызовутся друг за другом

```suggestion
            event.target.classList.add('refresh-button--loading'); // вешаем перед ассинхронной операцией
            getToDoFetch()
                .then((ToDo) => ToDo?.forEach(element => {
                    createToDo(element);
                } ) )
                .catch((error) => console.log(error))
                .finally(() => event.target.classList.remove('refresh-button--loading')) //снимаем когда операция отработала
```